netty-reactive-streams/src/main/java/com/typesafe/netty/HandlerPublisher.java

```diff
@@ -203,32 +203,40 @@ private void provideSubscriber(Subscriber<? super T> subscriber) {
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-        verifyRegisteredWithRightExecutor(ctx);
+        // If the channel is not yet registered, then it's not safe to invoke any methods on it, eg read() or close()
+        // So don't provide the context until it is registered.
+        if (ctx.channel().isRegistered()) {
+            provideChannelContext(ctx);
+        }
+    }
+
+    @Override
+    public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
+        provideChannelContext(ctx);
+        ctx.fireChannelRegistered();
```

Comment:
It looks like you're calling `ctx.fireChannelRegistered()` because that's the existing behaviour of the `ChannelDuplexHandler` we're overriding here. I think it might be clearer to say `super. channelRegistered(ctx)` instead of copying the superclass's behaviour into this method.